corehq/apps/change_feed/consumer/feed.py

```diff
@@ -57,27 +74,42 @@ def iter_changes(self, since, forever):
             assert not forever, 'Kafka pillow should not timeout when waiting forever!'
             # no need to do anything since this is just telling us we've reached the end of the feed
 
-    def get_latest_change_id(self):
-        consumer = self._get_consumer(MIN_TIMEOUT)
-        # we have to fetch one change to populate the highwater offset
+    def get_current_offsets(self):
+        consumer = self._get_consumer(MIN_TIMEOUT, auto_offset_reset='smallest')
         try:
-            consumer.next()
-        except (ConsumerTimeout, KafkaConfigurationError, KafkaUnavailableError) as e:
+            # we have to fetch the changes to populate the highwater offsets
+            # todo: there is likely a cleaner way to do this
+            changes = list(consumer)
+        except ConsumerTimeout:
+            pass
+        except (KafkaConfigurationError, KafkaUnavailableError) as e:
             # kafka seems to be having issues. log it and move on
-            logging.exception(u'Problem getting latest change form kafka for {}: {}'.format(
+            logging.exception(u'Problem getting latest offsets form kafka for {}: {}'.format(
                 self,
                 e,
             ))
             return None
-        return consumer.offsets('highwater')[(self._topic, self._partition)]
+
+        highwater_offsets = consumer.offsets('highwater')
```

Comment:
The latest version of the kafka lib seems to have a better API for getting this: http://kafka-python.readthedocs.org/en/master/apidoc/KafkaConsumer.html#kafka.KafkaConsumer.highwater